plugins/system_upgrade.py

```diff
@@ -688,12 +703,18 @@ def transaction_download(self):
         self.log_status(_("Download finished."), DOWNLOAD_FINISHED_ID)
 
     def transaction_upgrade(self):
-        Plymouth.message(_("Upgrade complete! Cleaning up and rebooting..."))
-        self.log_status(_("Upgrade complete! Cleaning up and rebooting..."),
-                        UPGRADE_FINISHED_ID)
+        if self.state.poweroff_after is True:
+            Plymouth.message(_("Upgrade complete! Cleaning up and powering off..."))
+            self.log_status(_("Upgrade complete! Cleaning up and powering off..."),
+                            UPGRADE_FINISHED_ID)
+        else:
+            Plymouth.message(_("Upgrade complete! Cleaning up and rebooting..."))
+            self.log_status(_("Upgrade complete! Cleaning up and rebooting..."),
+                            UPGRADE_FINISHED_ID)
+
```

Comment:
I would set just the message inside the above `if` block matching the reboot / poweroff intention. Then we could use the `Plymouth.message` and `self.log_status` call just once passing the message variable.